### PR TITLE
[jit] Call mono_class_setup_fields () before accessing field->offset. Fixes https://github.com/mono/mono/issues/17665.

### PR DESCRIPTION
**Pull request by vargaz**
Fixes https://github.com/mono/mono/issues/17665

### CODE REVIEW

**Comment by vargaz** [comment]
@monojenkins backport 2019-10

File: mono/mini/method-to-ir.c [hunk 1]
```diff
@@ -9139,6 +9140,8 @@ mono_method_to_ir (MonoCompile *cfg, MonoMethod *method, MonoBasicBlock *start_b
 
 			/* INSTANCE CASE */
 
+			if (is_instance)
+				g_assert (field->offset);
```
Comment (jaykrell) [inline mono/mini/method-to-ir.c]: `g_assert (!is_instance || field->offset);`

Comment (vargaz) [reply mono/mini/method-to-ir.c]: So the difference between the two is that the if version is easy to understand, while the other version requires some thinking.

Comment (jaykrell) [reply mono/mini/method-to-ir.c]: This basic boolean logic, alas.

**Pull request merged**